src/averageAges.js

```diff
@@ -52,7 +72,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const diff = people
+    .filter(person => {
+      return onlyWithSon
+        ? people.some(mother => person.mother === mother.name)
+          && person.sex === 'm'
+        : people.some(mother => person.mother === mother.name);
```

Comment:
you write the same iteration twice, could be assigned to a variable and used later